src/plugins/workspace/server/saved_objects/saved_objects_wrapper_for_check_workspace_conflict.ts

```diff
@@ -0,0 +1,257 @@
+            item.workspaces =
+              objectsMapWorkspaces[
+                this.getRawId({
+                  namespace,
+                  id: item.id,
+                  type: item.type,
+                })
+              ];
+          }
+
+          return item;
+        });
+      }
+      const realBulkCreateResult = await wrapperOptions.client.bulkCreate(objectsPayload, options);
+      const result: SavedObjectsBulkResponse = {
+        ...realBulkCreateResult,
+        saved_objects: [...objectsConflictWithWorkspace, ...realBulkCreateResult.saved_objects],
+      };
+      return result as SavedObjectsBulkResponse<T>;
+    };
+
+    const checkConflictWithWorkspaceConflictCheck = async (
+      objects: SavedObjectsCheckConflictsObject[] = [],
+      options: SavedObjectsBaseOptions = {}
+    ) => {
+      const objectsConflictWithWorkspace: SavedObjectsCheckConflictsResponse['errors'] = [];
+      if (options.workspaces) {
+        if (objects.length === 0) {
+          return { errors: [] };
+        }
+
+        const bulkGetDocs: any[] = objects.map((object) => {
+          const { type, id } = object;
+
+          return {
+            type,
+            id,
+            fields: ['id', 'workspaces'],
+          };
+        });
+
+        if (bulkGetDocs.length) {
+          const bulkGetResult = await wrapperOptions.client.bulkGet(bulkGetDocs);
+
+          bulkGetResult.saved_objects.forEach((object) => {
+            const { id, type } = object;
+            if (!object.error) {
+              let workspaceConflict = false;
+              if (options.workspaces) {
+                const filteredWorkspaces = SavedObjectsUtils.filterWorkspacesAccordingToSourceWorkspaces(
+                  options.workspaces,
+                  object.workspaces
+                );
+                if (filteredWorkspaces.length) {
+                  workspaceConflict = true;
+                }
+              }
+              if (workspaceConflict) {
+                objectsConflictWithWorkspace.push({
+                  id,
+                  type,
+                  error: {
+                    ...errorContent(SavedObjectsErrorHelpers.createConflictError(type, id)),
+                    metadata: { isNotOverwritable: true },
+                  },
+                });
+              }
+            }
+          });
+        }
+      }
+
+      const objectsFilteredByError = objects.filter(
+        (item) =>
+          !objectsConflictWithWorkspace.find(
+            (errorItems) =>
+              this.getRawId({
+                namespace: options.namespace,
+                type: errorItems.type,
+                id: errorItems.id,
+              }) ===
+              this.getRawId({
+                namespace: options.namespace,
+                type: item.type,
+                id: item.id as string,
+              })
+          )
+      );
+      const realBulkCreateResult = await wrapperOptions.client.checkConflicts(
```

Comment:
Shall we add empty check here?

Reply:
The real checkConflicts should handle the empty case, I'd keep as it is.